Updated issues templates

JasonStoltz opened the pull request:
## Summary

Provide a detailed summary of your PR. Explain how you arrived at your solution. If it includes changes to UI elements include a screenshot or gif.

If this is your first PR in the EUI repo, please ensure you've fully read through our [contributing to EUI](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui#how-to-ensure-the-timely-review-of-pull-requests) wiki guide.

## QA

Remove or strikethrough items that do not apply to your PR.

### General checklist

- Browser QA
    - [ ] Checked in both **light and dark** modes
    - [ ] Checked in both [MacOS](https://support.apple.com/lv-lv/guide/mac-help/unac089/mac) and [Windows](https://support.microsoft.com/en-us/windows/turn-high-contrast-mode-on-or-off-in-windows-909e9d89-a0f9-a3a9-b993-7a6dcee85025) **high contrast modes**
      - (_[emulate forced colors](https://devtoolstips.org/tips/en/emulate-forced-colors/) if you do not have access to a Windows machine_.)
    - [ ] Checked in **mobile**
    - [ ] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
    - [ ] Checked for **accessibility** including keyboard-only and screenreader modes
- Docs site QA
    - [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**
    - [ ] Props have proper **autodocs** (using `@default` if default values are missing) and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/playgrounds.md)**
    - [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples
- Code quality checklist
    - [ ] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/unit-testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/cypress-testing.md) tests**
    - [ ] Updated **[visual regression tests](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/visual-regression-testing.md)**
- Release checklist
    - [ ] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md)** entry exists and is marked appropriately.
    - [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)
- Designer checklist
  - [ ] If applicable, [file an issue](https://github.com/elastic/platform-ux-team/issues/new/choose) to update [EUI's Figma library](https://www.figma.com/community/file/964536385682658129) with any corresponding UI changes. _(This is an internal repo, if you are external to Elastic, ask a maintainer to submit this request)_

tkajtoch commented on .github/ISSUE_TEMPLATE/02-feature_request.md:
```diff
@@ -7,18 +7,42 @@ type: 'Enhancement'
 assignees: ''
 
 ---
+Please replace the template text in the sections below when filling out this request, to the best of your ability. It helps us prioritize work. Please don't delete sections. If you can't fill something out right now, just replace the text with "N/A".
```
I had this thought for a while... Why don't we put these template fragments in HTML comment tags so they provide context when creating an issue, but don't bloat it with unnecessary content when created?

```markdown
**Problem Statement**
<!-- What problem do we need to solve? Why is this needed? -->
```

tkajtoch reviewed (commented):
I really like this change. The template is simple to follow and provides more valuable context to help put features on our roadmap.

[slightly off topic]
The thing I'd like to clarify and put guidelines around beyond this template is how we approach quick support duty tasks. I like having history in the form of an issue but sometimes it might take more time to create it than to fix the actual thing someone requests. Is it worth it? Should we aim towards "all work must be represented in issues" kind of approach?

There are many places we could document why these small changes are needed - we have git commit messages, PR descriptions (that are converted to git merge messages), gh issues, and more. Without guidelines it's up to a developer to decide where to put this information, and looking at this project's history, we've lost a ton of context over the years because some things were never documented, and I'd like to avoid that moving forward.

JasonStoltz commented:
@tkajtoch Let me know what you think

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-deploy-docs/builds/315)
* Commit: 3f7e4c1c2d206675e693b8973d5487cd2e4586e1
* [Documentation website](https://eui.elastic.co/pr_8696/)


### History
* :green_heart: [Build #296](https://buildkite.com/elastic/eui-deploy-docs/builds/296) succeeded 9c515ef1476eba8cc9138bab83f949e9250338ab

<!--buildkite-pr-comment-eui-deploy-docs
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-deploy-docs/builds/315","number":315,"commit":"3f7e4c1c2d206675e693b8973d5487cd2e4586e1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-deploy-docs/builds/296","number":296,"commit":"9c515ef1476eba8cc9138bab83f949e9250338ab"}],"number":315}
buildkite-pr-comment-->

weronikaolejniczak commented:
@JasonStoltz do we want to merge this PR?

JasonStoltz commented:
@weronikaolejniczak Sorry, I was holding this because I had a small edit I wanted to make.

I had added the following section to the Feature Request template:

```
**Impact assessment (Optional)**

<!-- Just leave this section here and the EUI team will fill it out as needed. 

Examples:
- Is this a breaking API change?
- Are there structural or visual changes that would impact tests or CSS overrides?
- If either of those are true, please consider:
  - How many component usages in Kibana would this affect?
  - How much effort will it take consumers to adapt to this change?
  - How much risk is involved in making this change?

-->
```

Originally, I envisioned us (as maintainers) filling in sections like this in Feature Requests as we start planning and executing work on Feature Requests.

However, after some research, I think it's more typical to capture that information elsewhere, keeping the original Feature Request in tact.

So I removed that, and I am now merging this PR. Thanks.
